web/src/server_events_dispatch.js

```diff
@@ -1010,8 +1011,27 @@ export function dispatch_normal_event(event) {
                 case "update": {
                     const group_id = event.group_id;
                     const group = user_groups.get_user_group_from_id(group_id);
+
+                    const changed_group_settings = group_permission_settings
+                        .get_group_permission_settings()
+                        .filter((setting_name) => event.data[setting_name] !== undefined);
+                    const old_setting_values = {};
+                    for (const setting_name of changed_group_settings) {
+                        old_setting_values[setting_name] = group[setting_name];
+                    }
                     user_groups.update(event, group);
                     user_group_edit.update_group(event, group);
+
+                    for (const setting_name of changed_group_settings) {
+                        const $elem = $(
+                            `#id_group_permission_${CSS.escape(group_id.toString())}_${CSS.escape(setting_name)}`,
+                        );
+                        user_group_edit.update_setting_in_group_permissions_panel(
+                            $elem,
+                            group[setting_name],
+                            old_setting_values[setting_name],
+                        );
+                    }
```

Comment:
Should `user_group_edit.update_group` be responsible for this extra logic? We generally like to keep `server_events_dispatch` free of UI considerations like selectors.